jobs/build-charms/charms.py

```diff
@@ -269,18 +270,26 @@ def info(name, **query):
 
     def status(self, charm_entity):
         """Read CLI Table output from charmcraft status and parse."""
-        charm_status = self.charmcraft.status(charm_entity)
+        charm_status = self.charmcraft.status(charm_entity, _out=None)
         header, *body = charm_status.stdout.decode().splitlines()
         channel_status = self._table_to_list(header, body)
 
         for idx, row in enumerate(channel_status):
-            resources = row.get("Resources", "")
-            if resources == "-":
-                row["Resources"] = []
-            elif resources == "↑":
-                row["Resources"] = channel_status[idx - 1]["Resources"]
-            else:
-                row["Resources"] = dict(self.STATUS_RESOURCE.findall(resources))
+            base_split = self.BASE_RE.findall(row.get("Base"))
+            if base_split:
+                row["Base"] = dict(
+                    zip(["name", "channel", "architecture"], base_split[0])
+                )
+            for prop in ["Resources", "Revision", "Version"]:
+                value = row.get(prop, "")
+                if value == "↑":
+                    row[prop] = channel_status[idx - 1][prop]
+                elif value == "-":
+                    row[prop] = {} if prop == "Resources" else None
+                elif prop == "Resources":
+                    row[prop] = dict(self.STATUS_RESOURCE.findall(value))
+                else:
+                    row[prop] = value
```

Comment:
`charmcraft status` shows a lot more than it used to, so we're gonna make sure all the ↑ and - and such gets converted appropriately